Armor Upgrade Item Summary & Tooltip Fix

ian612 opened the pull request:
Fixes issue #1052, where the upgrade chat data was being gathered incorrectly from upgrade items. Also fixes an issue where the properties tooltip was displaying "undefined" for properties without one.

ian612 commented:
Tooltip bug this fixes shown below.

![image](https://github.com/foundryvtt-starfinder/foundryvtt-starfinder/assets/34078802/e414a640-3419-43b8-8d00-d79283de4b87)

LebombJames commented:
My concern with this is with how it relates to existing upgrades; some have `armorType` set, others `allowedArmorType` and others both. It seems `allowedArmorType` is the one more commonly used

ian612 commented:
What's the resolution for that? Go through and edit all the upgrades in the compendium to use `allowedArmorType`?

LebombJames commented:
As far as I can tell, on any item with both `armorType` and `allowedArmorType`, `allowedArmorType` is correct. If an item has only `armorType`, then *that* is correct. Therefore I think the way forward is to simply delete `armorType` on items with both, and migrate `armorType` to `allowedArmorType` on items with only `armorType`.

ian612 commented:
> As far as I can tell, on any item with both `armorType` and `allowedArmorType`, `allowedArmorType` is correct. If an item has only `armorType`, then _that_ is correct. Therefore I think the way forward is to simply delete `armorType` on items with both, and migrate `armorType` to `allowedArmorType` on items with only `armorType`. Migrate in the most basic definition of the word; just needs to be a simple compendium node script.

I've added a node script for this in https://github.com/foundryvtt-starfinder/foundryvtt-starfinder/pull/1182/commits/8efad9426ddaf9eeb8799570528f2d26bf10431a and updated the relevant compendium items.

LebombJames commented on src/module/actor/sheet/base.js:
```diff
@@ -1020,8 +1020,9 @@ export class ActorSheetSFRPG extends ActorSheet {
+            console.log(chatData.properties);
             chatData.properties.forEach(p => props.append(
-                `<span class="tag" data-tooltip="${p.tooltip || p.title}"><strong>${p.title ? p.title + ":" : ""} </strong>${p.name}</span>`
+                `<span class="tag" data-tooltip="${(p.tooltip || p.title) ? (p.tooltip || p.title) : ""}"><strong>${p.title ? p.title + ":" : ""} </strong>${p.name}</span>`
```
It'd be better to only add a tooltip.at all if there is a tooltip or title, rather than producing an empty tooltip

ian612 replied:
Fixed both items as suggested

LebombJames commented on src/module/actor/sheet/base.js:
```diff
@@ -1020,8 +1020,9 @@ export class ActorSheetSFRPG extends ActorSheet {
             Hooks.callAll("renderItemSummary", this, div, {}); // Event listeners need to be added to this HTML.
 
             const props = $(`<div class="item-properties"></div>`);
+            console.log(chatData.properties);
```
Errant logging

LebombJames commented on src/module/actor/sheet/base.js:
```diff
@@ -1020,9 +1020,20 @@ export class ActorSheetSFRPG extends ActorSheet {
-            chatData.properties.forEach(p => props.append(
-                `<span class="tag" data-tooltip="${p.tooltip || p.title}"><strong>${p.title ? p.title + ":" : ""} </strong>${p.name}</span>`
-            ));
+            console.log(chatData.properties);
```
You added it back! 😆
